Code/GraphMol/Descriptors/Property.cpp

```diff
@@ -0,0 +1,184 @@
+    case Property::NumAlpiphaticHeteroCycles: return "Number of Aliphatic Hetero Cycles";
+    case Property::NumSpiroAtoms: return "Number of Spiro Atoms";
+    case Property::NumBridgeheadAtoms: return "Number of Bridgehead Atoms";
+    case Property::AMW: return "Average Molecular WeightW";
+    case Property::LabuteASA: return "Labute Approximate Surface Area";
+      
+    default:
+        return "User defined property";
+  }
+}
+
+std::string Property::getVersionForProp(PropType prop) {
+  switch (prop) {
+    case Property::MW: return  exactmwVersion;
+    case Property::TPSA: return tpsaVersion;
+    case Property::ALOGP: return crippenVersion;
+    case Property::NumRotors: return NumRotatableBondsVersion;
+    case Property::MissingStereo: return "1.0";
+    case Property::LipinskiHBA: return "1.0";
+    case Property::LipinskiHBD: return "1.0";
+    case Property::HBA: return NumHBAVersion;
+    case Property::HBD: return NumHBDVersion;
+    case Property::NumHeteroAtoms: return NumHeteroatomsVersion;
+    case Property::NumAmideBonds: return NumAmideBondsVersion;
+    case Property::FractionCSP3: return FractionCSP3Version;
+    case Property::NumRings: return NumRingsVersion;
+    case Property::NumAromaticRings: return NumAromaticRingsVersion;
+    case Property::NumAliphaticRings: return NumAliphaticRingsVersion;
+    case Property::NumSaturatedRings: return NumSaturatedRingsVersion;
+    case Property::NumHeteroCycles: return NumHeterocyclesVersion;
+    case Property::NumSaturatedHeteroCycles: return NumSaturatedHeterocyclesVersion;
+    case Property::NumAromaticHeteroCycles: return NumAromaticHeterocyclesVersion;
+    case Property::NumAromaticCarboCycles: return NumAromaticCarbocyclesVersion;
+    case Property::NumSaturatedCarboCycles: return NumSaturatedCarbocyclesVersion;
+    case Property::NumAlpiphaticHeteroCycles: return NumAliphaticCarbocyclesVersion;
+    case Property::NumSpiroAtoms: return NumSpiroAtomsVersion;
+    case Property::NumBridgeheadAtoms: return NumBridgeheadAtomsVersion;
+    case Property::AMW: return amwVersion;
+    case Property::LabuteASA: return labuteASAVersion;
+      
+    default:
+        return "User defined property";
+  }
+}
+
+bool Property::isAdditive() const {
+  switch(m_proptype) {
+    case Property::AMW: 
+    case Property::LabuteASA:
+      return false;
+    case Property::User:
+      PRECONDITION(m_propfxn.get(), "Null Property fxn");
+      return m_propfxn->isAdditive();
+    default:
+      return true;
+  }
+}
+
+
+namespace {
+double numUnspecifiedStereoAtoms(const ROMol &mol) {
+  double res=0.;
+  for (ROMol::ConstAtomIterator atom = mol.beginAtoms(); atom != mol.endAtoms();
+       ++atom) {
+    if ((*atom)->hasProp(common_properties::_ChiralityPossible) &&
+        (*atom)->getChiralTag() == Atom::CHI_UNSPECIFIED)
+      res++;
+  }
+  return res;
+}
+}
+
+double Property::computeProperty(const ROMol &mol) const {
+  const bool ignoreCachedValues = true;
```

Comment:
Why would this be true?
If you really want to be able to ignore caching, then I'd add a `ignoreCachedValues` argument that defaults to false. Otherwise this should be "false"